### PR TITLE
feat: network badge on token details page

### PR DESCRIPTION
**Pull request by kayleegeorge**
added network badge to token details page for Polygon, Optimism, and Arbitrum AND change loading state bubble color

Screen Shot 2022-07-29 at 12 02 06 PM

### CODE REVIEW

File: src/components/Explore/TokenDetails/TokenDetail.tsx [hunk 1]
```diff
@@ -176,15 +178,34 @@ const TruncatedAddress = styled.span`
...
+  font-weight: 600;
+  font-size: 12px;
+  line-height: 12px;
+  color: ${({ theme, networkColor }) => networkColor ?? theme.textPrimary};
+  background-color: ${({ theme, backgroundColor }) => backgroundColor ?? theme.backgroundSurface};
+`
 
 export default function LoadedTokenDetail({ address }: { address: string }) {
   const theme = useTheme()
+  const NETWORK_BADGE_COLORS: Record<string, string> = {
```
Comment (kayleegeorge) [inline src/components/Explore/TokenDetails/TokenDetail.tsx:195]: this is slightly ugly, lmk if you have ideas on how to do this better (not sure how to 'access' theme outside CSS/component)

File: src/components/Explore/TokenDetails/TokenDetail.tsx [hunk 2]
```diff
@@ -176,6 +177,17 @@ const TruncatedAddress = styled.span`
     display: flex;
   }
 `
+const NetworkBadge = styled.div<{ networkColor?: string; backgroundColor?: string }>`
+  display: flex;
```
Comment (vm) [inline src/components/Explore/TokenDetails/TokenDetail.tsx:181]: rm flex

File: src/components/Explore/TokenDetails/TokenDetail.tsx [hunk 3]
```diff
@@ -176,6 +177,17 @@ const TruncatedAddress = styled.span`
...
+  border-radius: 5px;
+  padding: 4px 6px;
+  align-items: center;
```
Comment (vm) [inline src/components/Explore/TokenDetails/TokenDetail.tsx:184]: rm

File: src/components/Explore/TokenDetails/TokenDetail.tsx [hunk 4]
```diff
@@ -176,6 +177,17 @@ const TruncatedAddress = styled.span`
...
+  font-weight: 600;
+  font-size: 12px;
+  line-height: 12px;
```
Comment (vm) [inline src/components/Explore/TokenDetails/TokenDetail.tsx:187]: isn't line height the same as font-size by default

File: src/components/Explore/TokenDetails/TokenDetail.tsx [hunk 5]
```diff
@@ -176,6 +177,17 @@ const TruncatedAddress = styled.span`
     display: flex;
   }
 `
+const NetworkBadge = styled.div<{ networkColor?: string; backgroundColor?: string }>`
+  display: flex;
+  border-radius: 5px;
+  padding: 4px 6px;
```
Comment (vm) [inline src/components/Explore/TokenDetails/TokenDetail.tsx:183]: 6px seems bad

File: src/components/Explore/TokenDetails/TokenDetail.tsx [hunk 6]
```diff
@@ -185,6 +197,9 @@ export default function LoadedTokenDetail({ address }: { address: string }) {
   const [activeTimePeriod, setTimePeriod] = useState(TimePeriod.hour)
   const isFavorited = favoriteTokens.includes(address)
   const toggleFavorite = useToggleFavorite(address)
+  const chainInfo = getChainInfo(token?.chainId)
+  const networkLabel = chainInfo?.label
+  const networkBadgeBackgroundColor = chainInfo?.background_color
```
Comment (vm) [inline src/components/Explore/TokenDetails/TokenDetail.tsx:202]: `backgroundColor`

File: src/components/Explore/TokenDetails/TokenDetail.tsx (hunk 4 above)
Comment (kayleegeorge) [reply src/components/Explore/TokenDetails/TokenDetail.tsx:187]: oh ya whoops lol

File: src/theme/colors.ts [hunk 7]
```diff
@@ -215,6 +217,10 @@ export interface Palette {
   chain_42161: Color
   chain_421611: Color
   chain_80001: Color
+
```
Comment (kayleegeorge) [inline src/theme/colors.ts:220]: rename to chain_n

**Review by vm — APPROVED** [review]

**Pull request merged**